Zotero/Scenes/Detail/ItemDetail/Views/ItemDetailViewController.swift

```diff
@@ -142,63 +197,65 @@ final class ItemDetailViewController: UIViewController {
             }
 
         case .openTagPicker:
-            self.coordinatorDelegate?.showTagPicker(libraryId: self.viewModel.state.library.identifier,
-                                                    selected: Set(self.viewModel.state.tags.map({ $0.id })),
-                                                    picked: { [weak self] tags in
-                                                        self?.viewModel.process(action: .setTags(tags))
-                                                    })
+            coordinatorDelegate?.showTagPicker(libraryId: viewModel.state.library.identifier, selected: Set(viewModel.state.tags.map({ $0.id })), picked: { [weak self] tags in
+                self?.viewModel.process(action: .setTags(tags))
+            })
 
         case .openTypePicker:
-            self.coordinatorDelegate?.showTypePicker(selected: self.viewModel.state.data.type,
-                                                     picked: { [weak self] type in
-                                                         self?.viewModel.process(action: .changeType(type))
-                                                     })
+            coordinatorDelegate?.showTypePicker(selected: viewModel.state.data.type, picked: { [weak self] type in
+                self?.viewModel.process(action: .changeType(type))
+            })
 
         case .openUrl(let string):
-            if let url = URL(string: string) {
-                self.coordinatorDelegate?.show(url: url)
-            }
+            guard let url = URL(string: string) else { return }
+            coordinatorDelegate?.show(url: url)
 
         case .openDoi(let doi):
             guard let encoded = FieldKeys.Item.clean(doi: doi).addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) else { return }
-            self.coordinatorDelegate?.show(doi: encoded)
+            coordinatorDelegate?.show(doi: encoded)
         }
     }
 
-    private func cancelEditing() {
-        self.viewModel.process(action: .cancelEditing)
-    }
-
     // MARK: - UI state
 
     /// Update UI based on new state.
     /// - parameter state: New state.
     private func update(to state: ItemDetailState) {
         if state.hideController {
-            self.navigationController?.popViewController(animated: true)
+            navigationController?.popViewController(animated: true)
             return
         }
 
         if let error = state.error {
-            self.coordinatorDelegate?.show(error: error, viewModel: self.viewModel)
+            coordinatorDelegate?.show(error: error, viewModel: viewModel)
         }
 
         if state.changes.contains(.item) {
+            // 🍎 check why this is done
```

Comment:
This state change calls `viewModel.process(action: .reloadData)` which causes another state change, which causes reentrancy issue in `RxSwift` (withing a subscription event you cause another subscription event). So I just `async` this so that it's called later. Didn't find a better way to deal with this.